### PR TITLE
Reenabling ts lint

### PR DESCRIPTION
**Pull request by mrpatpat**
Merge permission: {CC-Member}

## Description

- reenabling ts-lint
- refactoring rules
- refactoring code that fails rules
- removing ts-lint from webpack-loader. It is now a separate script. 

To prevent merge conflicts this should be merged after merging #274 and before starting #178.

depends on #274
blocks #178

## Definition of Done

A task/pull request will not be considered to be complete until all these items can be checked off.

* [ ] **All** requirements mentioned in the issue are implemented
* [x] Does match the Code of Conduct and the Contribution file 
* [ ] Task has its own **GitHub issue** (something it is solving)
  * [ ] Issue number is **included in the commit messages** for traceability
* [ ] **Update the README.md** with any changes/additions made
* [x] **Update the CHANGELOG.md** with any changes/additions made
* [ ] **Enough test coverage to ensure that uncovered changes do not break functionality**
* [x] **All tests pass**    
* [x] **Descriptive pull request text**, answering:
  + What problem/issue are you fixing?
  + What does this PR implement and how? 
* [x] **Assign your PR to someone** for a code review
  + This person _will be contacted **first**_ if a bug is introduced into `master`
* [x] **Manual testing** did not fail

### CODE REVIEW

File: visualization/tslint.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
     "rules": {
         "class-name": true,        
         "no-var-keyword": true,
+        "no-empty": [true, "allow-empty-catch", "allow-empty-functions"],
```
Comment (NearW) [inline visualization/tslint.json]: allow-empty-catch can be removed since catch-statements are optional anyway

File: visualization/tslint.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
         "no-eval": true,
         "no-conditional-assignment": true,
         "new-parens": true,
-        "no-unused-variable": true,
+        "no-unused-variable": [true, {"ignore-pattern": "^_"}],
```
Comment (NearW) [inline visualization/tslint.json]: underscore => variable is used in template and shouldn't be deleted.
make severity to warning.
Conventin should be noted somewhere

File: visualization/tslint.json [hunk 3]
```diff
@@ -9,6 +9,7 @@
         "no-conditional-assignment": true,
         "new-parens": true,
         "no-unused-variable": [true, {"ignore-pattern": "^_"}],
+        "member-access": true,
```
Comment (NearW) [inline visualization/tslint.json]: functions with no member-access are declared public now (they were public by default before anyway). Inform team to check files from time to time if some functions can be declared private

File: visualization/tslint.json [hunk 4]
```diff
@@ -10,6 +10,7 @@
         "new-parens": true,
         "no-unused-variable": [true, {"ignore-pattern": "^_"}],
         "member-access": true,
+        "member-ordering": [true, {"order": "instance-sandwich"}],
```
Comment (NearW) [inline visualization/tslint.json]: remove public->protected->private ordering. Functions should be declared right after they're called

File: visualization/app/codeCharta/core/download/download.service.ts [hunk 5]
```diff
@@ -61,11 +61,12 @@ export class DownloadService {
     }
 
     private downloadData(data, fileName) {
+        let d = data;
```
Comment (mrpatpat) [inline visualization/app/codeCharta/core/download/download.service.ts]: dataJson

**Comment by mrpatpat** [comment]
> I like the changes! but it will definitely result in some merge conflicts with the _markingColor-branch_.
> -> The LegendPanel doesn't show all the metrics anymore. Feature or bug?
> 

definetly not a feature ! 👍

File: visualization/tslint.json (hunk 2 above)
Comment (alschmut) [reply visualization/tslint.json]: See [wiki](https://github.com/MaibornWolff/codecharta/wiki/to-be-documented)

File: visualization/tslint.json (hunk 3 above)
Comment (alschmut) [reply visualization/tslint.json]: See [wiki](https://github.com/MaibornWolff/codecharta/wiki/to-be-documented)

**Pull request merged**